### PR TITLE
Differentiate namespaces in IIS functionals

### PR DESCRIPTION
**Pull request by HaoK**
Adds IISExpress/NewHandler/NewShim so the namespaces of the common tests display which variant failed

### CODE REVIEW

**Comment by HaoK** [comment]
These conditionals should also allow conditional quarantining of tests now too which might be a handy side perk if we don't want to quarantine tests across all variants

**Review by Tratcher — APPROVED** [review]

File: src/Servers/IIS/IIS/test/IISExpress.FunctionalTests/InProcess/AuthenticationTests.cs [hunk 1]
```diff
@@ -9,7 +9,7 @@
 using Microsoft.AspNetCore.Testing;
 using Xunit;
 
-namespace Microsoft.AspNetCore.Server.IIS.FunctionalTests
+namespace Microsoft.AspNetCore.Server.IIS.FunctionalTests.IISExpress
```
Comment (Tratcher) [inline src/Servers/IIS/IIS/test/IISExpress.FunctionalTests/InProcess/AuthenticationTests.cs:12]:
```suggestion
namespace Microsoft.AspNetCore.Server.IIS.IISExpress.FunctionalTests
```

**Comment by halter73** [comment]
It might be interesting to do the same thing for Kestrel's shared Sockets/Libuv Functional and "Bind" tests like the ones in https://github.com/dotnet/aspnetcore/tree/fa9dd9c5fe476706e33819ded8b96bf72a5fef41/src/Servers/Kestrel/test/FunctionalTests. It could be a follow up if we think it's worthwhile. [Libuv is scheduled to be removed in .NET 6 anyway](https://docs.microsoft.com/en-us/dotnet/core/compatibility/aspnet-core/5.0/kestrel-libuv-transport-obsolete), but I'm not 100% sure that's happening.

**Review by halter73 — APPROVED** [review]

File: src/Servers/IIS/IIS/test/Common.FunctionalTests/WindowsAuthTests.cs [hunk 2]
```diff
@@ -10,7 +10,20 @@
...
 
+#if !IIS_FUNCTIONALS
+using Microsoft.AspNetCore.Server.IIS.FunctionalTests;
+
+#if IISEXPRESS_FUNCTIONALS
+namespace Microsoft.AspNetCore.Server.FunctionalTests.IISExpress
+#elif NEWHANDLER_FUNCTIONALS
+namespace Microsoft.AspNetCore.Server.IIS.FunctionalTests.NewHandler
+#elif NEWSHIM_FUNCTIONALS
+namespace Microsoft.AspNetCore.Server.IIS.FunctionalTests.NewShim
```
[the remark this answers is not in the document]
Comment (HaoK) [reply src/Servers/IIS/IIS/test/Common.FunctionalTests/WindowsAuthTests.cs:21]: I followed the existing naming conventions which were prob chosen to save some usings, but I went ahead and updated all of them now to match the project names

File: src/Servers/IIS/IIS/test/Common.FunctionalTests/FrebTests.cs [hunk 3]
```diff
@@ -36,7 +49,6 @@ public static IList<FrebLogItem> FrebChecks()
         }
 
         [ConditionalFact]
-        [MaximumOSVersion(OperatingSystems.Windows, WindowsVersions.Win10_20H2, SkipReason = "Shutdown hangs https://github.com/dotnet/aspnetcore/issues/25107")]
```
Comment (Tratcher) [inline src/Servers/IIS/IIS/test/Common.FunctionalTests/FrebTests.cs:39]: ?

Comment (HaoK) [reply src/Servers/IIS/IIS/test/Common.FunctionalTests/FrebTests.cs:39]: Yeah I was using this branch to test stuff, I thought i was done, so push had some junk, reverted now

**Pull request merged**